File: cpp/228.cpp
#include<vector>
#include<string>
#include<utility>
using namespace std;

class Solution {
public:
  vector<string> summaryRanges(vector<int>& nums) {
    vector<pair<int, int>> rg;
    for(int n : nums)
      if(!rg.empty() && rg[rg.size()-1].second+1 == n)
        ++rg[rg.size()-1].second;
      else
        rg.push_back({n, n});
    vector<string> res;
    for(auto p : rg){
      int i=p.first, j=p.second;
      if(i == j)
        res.push_back(to_string(i));
      else
        res.push_back(to_string(i)+"->"+to_string(j));
    }
    return res;
  }
};

File: python/228.py
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        res = []
        for n in nums:
            if res and res[-1][1] + 1 == n:
                res[-1][1] += 1
            else:
                res.append([n, n])
        return [("{}->{}".format(i, j) if i != j else str(i))
                for i, j in res]
